docs/src/proposals/interest-bearing-tokens.md

```diff
@@ -0,0 +1,284 @@
+---
+title: Interest-Bearing Tokens
+---
+
+This document describes a solution for interest-bearing tokens as an on-chain
+program. Interest-bearing tokens continuously accrue interest as time passes.
+
+## Problem
+
+Yield on blockchain assets is a crucial feature of many decentralized finance
+applications, as token holders want to be compensated for providing their tokens
+as liquidity to applications. The concept of "yield farming" on tokens has even
+spawned very successful tongue-in-cheek projects, such as
+[$MEME farming](https://dontbuymeme.com/).
+
+Tokens that accrue interest are an attractive and popular version of yield farming,
+as seen most prominently with Aave's [aTokens](https://aave.com/atokens). For
+simplicity, we'll refer to interest-bearing tokens as i-tokens.
+
+### How do these work?
+
+Let's explain through an example: assume there's an i-token mint with a total
+supply of 1,000 tokens
+and a 10% annual interest rate. This means that one year later, the total supply
+will be 1,100 tokens, and all token holders will have 10% more tokens in their
+holding account.
+
+Since interest is accrued continuously, however, just 1 second later, the total
+supply has again increased. A 10% annualized rate gives us a continuous interest
+rate of ~9.5% (`ln(1.1)`), which allows us to calculate the new instantaneous supply.
+
+```
+new_supply = previous_supply * exp(continuous_rate / seconds_per_year)
+new_supply = 1100 * exp(0.095 / 31536000)
+new_supply = 1100.0000033...
+```
+
+Since i-tokens are constantly accruing interest, they do not have the same
+properties as normal tokens. Most importantly, i-tokens held in two different
+accounts are only fungible if they have been updated to the same time. For example,
+10 i-tokens in the year 1999 are worth much more than 10 i-tokens in the year 3000, since
+those tokens from 1999 have over 1,000 years of interest (see Futurama). Because
+of this, we have to compound interest before any minting, burning, and
+transferring.
+
+### Initial Attempt
+
+The concept seems simple to implement. We can create an SPL token, allow
+tokens to be minted by a new "interest accruing" program, and then make sure to
+call it regularly.
+
+This solution has a few problems. First, how do we compound interest regularly?
+We can create a separate service which periodically sends transactions to update
+all of the accounts. Unfortunately, this will cost the service a lot, especially if
+there are thousands of accounts to update.
+
+We can enforce compounding before any transfer / mint / burn, but then all programs
+and web consumers need to change to make these update calls. How do we change programs
+and wallets to properly update accounts without creating a big `if is_i_token` case
+all over the code?
+
+This solution creates friction for almost all SPL token users: on-chain programs,
+wallets, applications, and end-users.
+
+## Proposed Solution
```

Comment:
a bit lazy question, but how our https://github.com/solana-labs/solana-program-library/tree/90b4b33775b0a464ca443963e65ebc80756a3e2e/token-lending is working today to pay interests from lending without this i-token is still in proposal stage?

Reply:
No worries!  In its current form, the liquidity providers for token lending get yield directly from the borrowers in the form of repayments.  So if you deposit USDC into the liquidity pool, and no one borrows USDC, then you don't get anything.  Maybe we should think about that as more of a liquidity provider fee?

On the other hand, the i-token generates a constant interest like a savings account with a bank, so you always get a fixed percentage gain.